packages/main/src/plugin/configuration-impl.ts

```diff
@@ -54,14 +55,70 @@ export class ConfigurationImpl implements containerDesktopAPI.Configuration {
+    //
+    // Locked keys ignore user settings and force the managed default value instead, if there is a value there
+    // in the first place, if not, we fall back to the user value.
+    //
+    // We retrieve the information from from a system file:
+    // (ex macOS: /Library/Application Support/com.podman.desktop/locked.json, Linux: /usr/share/podman-desktop/locked.json)
+    //
+    // This only applies when we are getting values from the default scope, as getting them SPECIFICALLY
+    // from managed defaults or locked scopes would be counter-intuitive... (or Onboarding scope, but that doesn't make sense either)
+    if (this.scope === CONFIGURATION_DEFAULT_SCOPE) {
+      // Pass in the key we are wanting to get and check locked config for the key
+      const managedValue = this.getLockedConfigValue<T>(localKey);
+
+      // Before we go ahead and return the value, we want to log to console
+      // this information so that if users are confused why they can't change a setting, they know why,
+      // however, this information should only stay in console.debug considering that it could be noisy (returning configuration values multiple times)
+      //
+      // A GUI indication will be provided in the settings page for locked values:
+      // https://github.com/podman-desktop/podman-desktop/issues/13900
+      if (managedValue !== undefined) {
+        console.debug(
+          `[Managed-by]: Configuration key '${localKey}' is locked. Returning managed default value instead of user value.`,
+        );
+        return managedValue;
+      }
+    }
+
+    // If the above isn't applicable (no locked keys + managed by scope), we return the key as normal.
     const localView = this.getLocalView();
     if (localView[localKey] !== undefined) {
       return localView[localKey] as T;
     }
+
+    // Last resort: Return the fallback default value
     return defaultValue as T;
   }
 
+  // Helper to check if a config key is locked and return its managed value
+  // Returns undefined if the key isn't locked or has no managed value
+  private getLockedConfigValue<T>(localKey: string): T | undefined {
+    const lockedConfig = this.configurationValues.get(CONFIGURATION_SYSTEM_MANAGED_LOCKED_SCOPE);
+
+    // Bail early if there's no locked config or it's malformed
+    if (!lockedConfig?.['locked'] || !Array.isArray(lockedConfig['locked'])) {
+      return undefined;
+    }
+
+    const lockedKeys = lockedConfig['locked'] as string[];
+
+    // Bail early if this key isn't in the locked list
+    if (!lockedKeys.includes(localKey)) {
+      return undefined;
+    }
+
+    // This key IS locked - grab the managed default value for it
+    const managedDefaults = this.configurationValues.get(CONFIGURATION_SYSTEM_MANAGED_DEFAULTS_SCOPE);
+    if (managedDefaults?.[localKey] !== undefined) {
+      return managedDefaults[localKey] as T;
+    }
+
+    return undefined;
+  }
+
```

Comment:
it should not be there, it should be in a LockConfiguration object

const managedValue = this.getLockedConfigValue<T>(localKey);
-->

this.lockedConfiguration.get(...) and it's in this class that you can write comments about lock configuration